assessments/0011-keda/keda-analysis.md

```diff
@@ -0,0 +1,416 @@
+
+**Out of scope:**
+- Other KEDA repos: https://github.com/kedacore/*
+
+## How this document is organized
+
+This document is divided into three sections that represent three major areas of concern:
+
+- **Project documentation:** concerns documentation for users of the KEDA software, aimed at people who intend to use it
+- **Contributor documentation:** concerns documentation for new and existing contributors to the KEDA OSS project
+- **Website:** concerns the mechanics of publishing the documentation, and includes branding, website structure, and maintainability
+
+Each section begins with summary ratings based on a rubric with appropriate [criteria][cncf-doc-criteria] for the section, then proceeds to:
+- **Comments**: observations about the existing documentation, with a focus on how it does or does not help KEDA users achieve their goals.
+- **Recommendations**: suggested changes that would improve the effectiveness of the documentation. 
+
+An accompanying document, [keda-implementation.md][implementation-doc], breaks the recommendations down into concrete actions that can be implemented by project contributors. Its focus is on drilling down to specific, achievable work that can be completed in constrained blocks of time. Ultimately, the implementation items should be tracked as a series of Github [issues][keda-issues].
+
+
+## How to use this document
+
+Readers interested only in actionable improvements should skip this document and read [keda-implementation.md](./keday-implementation.md).
+
+Readers interested in the current state of the documentation and the reasoning behind the recommendations should read the section of this document pertaining to their area of concern:
+
+- [Project documentation](#project-documentation)
+- [Contributor documentation](#contributor-documentation)
+- [Website and documentation infrastructure](#website)
+
+Examples of CNCF documentation that demonstrate the analysis criteria are linked from the [Criteria][cncf-doc-criteria] specification. 
+
+
+### Recommendations, requirements, and best practices
+
+This analysis measures documentation against CNCF project maturity standards, and suggests possible improvements. In most cases there is more than one way to do things. Few recommendations here are meant to be prescriptive. Rather, the recommended implementations represent the reviewers' experience with how to apply documentation best practices. In other words, borrowing terminology from the lexicon of [RFCs][rfc-keywords], the changes described here should be understood as "recommended" or "should" at the strongest, and "optional" or "may" in many cases. Any "must" or "required" actions are clearly denoted as such, and pertain to legal requirements such as copyright and licensing issues. 
+
+
+# Project documentation
+
+KEDA is a **graduated** project of CNCF. This means that the project should have [*very high*][cncf-doc-criteria] standards for documentation.
```

Comment:
This is interesting, we should have better docs based on this report; but why is this not part of CNCF Graduation criteria? This report is valuable

Reply:
@tomkerkhove Glad you think the report is valuable, thanks! 😁 

I had a quick look for anything that might suggest a doc requirement for overall project maturity. There's this from the Application Process Principles in the TOC's [CNCF Graduation Application Template](https://github.com/cncf/toc/blob/main/.github/ISSUE_TEMPLATE/template-graduation-application.md):

"Additional documentation as appropriate for project type, e.g.: installation documentation, end user documentation, reference implementation and/or code samples." 

Of course, that's pretty vague. 

The evaluation criteria used for the analysis are [here](https://github.com/cncf/techdocs/blob/main/assessments/criteria.md). The criteria document lists the properties to be evaluated, but does not key their quality or presence to the project maturity level.

@nate-double-u , Are there other guidelines or criteria that I've missed?

I agree that the criteria could be upgraded. Imposing a requirement on CNCF should be a community decision, but I can work to update the doc criteria themselves if you have suggestions.

Reply:
> I had a quick look for anything that might suggest a doc requirement for overall project maturity. There's this from the Application Process Principles in the TOC's [CNCF Graduation Application Template](https://github.com/cncf/toc/blob/main/.github/ISSUE_TEMPLATE/template-graduation-application.md?rgh-link-date=2024-03-08T18%3A24%3A20Z):

Which is a new template and vague indeed :) I think this did not come up because it was not there when we went to Graduation which is OK.

That makes sense, thanks for clarifying!